modules/core/shared/src/main/scala/scaladex/core/model/search/ProjectDocument.scala

```diff
@@ -18,6 +18,7 @@ final case class ProjectDocument(
     organization: Project.Organization,
     repository: Project.Repository,
     artifactNames: Seq[Artifact.Name],
+    deprecatedArtifactNames: Seq[Artifact.Name],
```

Comment:
Because the search API is based on Elasticsearch, we need to add `deprecatedArtifactNames` field in the Elasticsearch schema.

The search documents are automatically updated by the `SearchSynchronizer` background job.